src/components/ProductFilters.tsx

```diff
@@ -1,103 +1,117 @@
-import { ChangeEvent, FC, useEffect, useState } from "react";
+import { ChangeEvent, FC, useState } from "react";
 import { useAppDispatch, useTypedSelector } from "../store";
 import {
-  filterProductsByPrice,
-  filterProductsBySize,
-  sortProductsByType,
-  updateChosenSizes,
-} from "../store/slices/productList.slice";
+  setSizeFilter,
+  setPriceFilter,
+  setSortType,
+  sortByTypeAction,
+  filterByPriceAction,
+  filterBySizeAction,
+} from "../store/slices/productFilters";
 import { setGloablLoading } from "../store/slices/global.slice";
 import Form from "react-bootstrap/Form";
-import { Col, Container } from "react-bootstrap";
+import { Button, Col, Container, Row } from "react-bootstrap";
+import { singleProduct } from "../mock-data";
+import { productSortTypes } from "../store/slices/productFilters/model";
 
-const selectSortOptions = [
-  { type: "low_First", name: "low price first" },
-  { type: "high_First", name: "high price first" },
-];
 export type ProductFiltersProps = {};
 
 const ProductFilters: FC<ProductFiltersProps> = () => {
-  const filter = useTypedSelector((state) => state.productList.productFilters);
-  const [maxPrice, setMaxPrice] = useState<number>(filter.price.max);
   const dispatch = useAppDispatch();
+  const {filterOptions} = useTypedSelector((state) => state.productFilters);
+  const [activeButton, setActiveButton] = useState<boolean>(false);
+  const [maxPrice, setMaxPrice] = useState<number>(100);
+  const [minPrice, setMinPrice] = useState<number>(1);
 
-  const handleChangeSort = (e: ChangeEvent<HTMLSelectElement>) => {
-    dispatch(sortProductsByType(e.target.value));
-  };
-  const handleChandeMaxPrice = (e: ChangeEvent<HTMLInputElement>) => {
-    const currMaxPriceValue = Number(e.target.value);
-    if (currMaxPriceValue < 1) return;
-    setMaxPrice(currMaxPriceValue);
+  const handlePriceUpdate = (e: ChangeEvent<HTMLInputElement>) => {
+    const valueMax = Number(e.target.value);
+    setMaxPrice(valueMax);
+    // setMinPrice(value)
+    setActiveButton(true);
```

Comment:
Suggestion: maybe we can make it with 'useEffect' that watching by 'filterOptions' object (set 'activeButton' inside hook). I'm not sure.
And remove setActiveButton from each handler.

Reply:
Done!